.github/workflows/cs.yaml

```diff
@@ -7,14 +7,18 @@ on:
       - main
       - master
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
```

Comment:
It makes total sense for a proprietary project as you pay for every minute, but it can hurt visibility for open-source projects. What are your arguments pro and contra?

Reply:
I recently contributed to this project (#1978), and each commit triggered the CI pipeline. The CI can take a bit long (up to 30mn for E2E tests).

GitHub allows Open Source projects to use its GitHub-hosted runners for free, but with a limit of runners and CPU. That means while the E2E tests are still running from a previous commit, if I push a new commit (and the number of GitHub runners allowed is reached), I'll have to wait longer for the new CI to start.

Enabling concurrency automatically cancels previous CI runs for the same PR/branch (with this configuration) thanks to `cancel-in-progress: true`.

Reply:
I would also not care if the tests were super fasts, but we are hurting for bandwidth, 1 or 2 PR and we block the whole pipeline for 30min - 1h.

I would also say it's good to save needless computation time, I'd wager it's very rare that you need the CI of past commits to carry on